torch/csrc/jit/source_range.h

```diff
@@ -78,37 +90,51 @@ struct Source {
   std::string text_;
   c10::optional<std::string> filename_;
   // If filename_ is not present, starting_line_no_ is don't care
-  size_t starting_line_no_;
+  int64_t starting_line_no_;
   // Starting offsets for lines into the source. e.g. line 0 starts at
   // line_starting_offsets_[0], etc.
-  std::vector<size_t> line_starting_offsets_;
+  std::vector<int64_t> line_starting_offsets_;
+  // serialized IValue representing this Source
+  // Lazily populated, nullptr if not populated
+  //
+  // NB: if you introduce methods that mutate a Source object, you *must*
```

Comment:
I don't think we can trust that people will read this comment, so it will probably just be a source of bugs. I am not a fan of storing the same information twice in a class for this reason. I think it would be better for this logic to be self-contained in a SourceSerializer object which can manage de-deduping the serialization state.